corehq/apps/user_importer/importer.py

```diff
@@ -502,7 +502,7 @@ def create_or_update_commcare_users_and_groups(upload_domain, user_specs, upload
                 else:
                     status_row['flag'] = 'created'
 
-                if phone_numbers:
+                if type(phone_numbers) is list:
```

Comment:
this would always be a list, no?

Reply:
@mkangia It's supposed to, yes. The reason for this...obvious check...is because I want to continue execution only when it's a list even if it's empty, but the truthiness of `[]` is False. So simply checking for a list-like object solves both problems (only list and empty list).

Reply:
[Line 480](https://github.com/dimagi/commcare-hq/blob/187ae4ca2bba47a9b7863c19a14db6edf94afdd7/corehq/apps/user_importer/importer.py#L480) is confusing:
```python
            phone_numbers = row.get('phone-number', []) if 'phone-number' in row else None
```
If "phone-number" is not in `row`, then `phone_numbers` will be `None`, not `[]`, because that `get()` method's default value will never get used.

You could change that line to
```python
            phone_numbers = row.get('phone-number') or []
```
if you want to avoid `None`s. But if it is possible for `row['phone-number']` to be something other than a list (is it possible? what makes it a list?), it might make sense to leave this `if` clause here.

Reply:
@Charl1996 you could say `if phone_numbers is not None` or simply `if 'phone-number' in row`

@kaapstorm I think what Charl needs is to avoid doing any processing on phone numbers if they are not passed, hence its assigned to `None` if column is missing but ya that default wont really get used.
So it is same as `phone_numbers = row['phone-number'] if 'phone-number' in row else None`

Reply:
> But if it is possible for row['phone-number'] to be something other than a list

I think the parsing logic ensures that it is going to be an empty list if nothing is passed but Charl could confirm if there is no way for this to be anything other than a list. He said this should be a list above.
May be could add a test to pass an empty string, assuming that would be the value if nothing is set? Currently the tests are using what is passed after parsing.

Reply:
@mkangia This discussion made me realise that the following would work best 
> phone_numbers = row.get('phone-number', [])

because when no phone-number columns are specified in the upload file, the user phone numbers should also be removed. Since no `phone-number` key will be specified in the row in the case that no phone number columns are specified, it would have resulted in `None` which would have skipped the check when it should not have. 

I'm pushing a new commit with a test for no phone-number in row.

Reply:
> because when no phone-number columns are specified in the upload file, the user phone numbers should also be removed.

can we confirm this with product? If we support partial uploads, where a user can only columns they want an action on, this would conflict with that. 
My vote would be to clear out phone numbers only if user explicitly tells HQ to, which I guess would be by passing an empty column, though I don't fully understand that width of upload feature to suggest best way forward, hence product.

Reply:
@mkangia You're right. I didn't even think of partial uploads. I'll quickly follow up on whether we support that or not.

Reply:
@mkangia Partial updates seems to be a thing. Well spotted!